src/keyboard.rs

```diff
@@ -648,40 +734,20 @@ pub enum KeyCode {
     F35,
 }
 
-/// Key represents the meaning of a keypress.
+/// Action represents the meaning of a keypress.
 ///
 /// This mostly conforms to the UI Events Specification's [`KeyboardEvent.key`] with a few
 /// exceptions:
 /// - The `Super` variant here, is named `Meta` in the aforementioned specification. (There's
 ///   another key which the specification calls `Super`. That does not exist here.)
 /// - The `Space` variant here, can be identified by the character it generates in the
 ///   specificaiton.
-/// - The `Unidentified` variant here, can still identifiy a key through it's `NativeKeyCode`.
-/// - The `Dead` variant here, can specify the character which is inserted when pressing the
-///   dead-key twice.
 ///
 /// [`KeyboardEvent.key`]: https://w3c.github.io/uievents-key/
 #[non_exhaustive]
-#[derive(Debug, Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, PartialOrd, Ord)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-pub enum Key<Str = SmolStr> {
-    /// A key string that corresponds to the character typed by the user, taking into account the
-    /// user’s current locale setting, and any system-level keyboard mapping overrides that are in
-    /// effect.
-    Character(Str),
-
-    /// This variant is used when the key cannot be translated to any other variant.
-    ///
-    /// The native key is provided (if available) in order to allow the user to specify keybindings
-    /// for keys which are not defined by this API, mainly through some sort of UI.
-    Unidentified(NativeKey),
-
-    /// Contains the text representation of the dead-key when available.
-    ///
-    /// ## Platform-specific
-    /// - **Web:** Always contains `None`
-    Dead(Option<char>),
-
+pub enum Action {
```

Comment:
It's not a `Named` key: that's what `KeyCode`s are.

No, I'm not convinced that `Action` is the best name either.

Reply:
The `Named` was suggested because of https://www.w3.org/TR/uievents-key/#named-key-attribute-value . Since they refer to categories under `named`, and given that you basically merged all categories it's named in w3c spec which winit follows.

I just don't like `Action` as well....

Reply:
>It's not a Named key: that's what KeyCodes are.

it's actually not, `KeyCodes` have regular `A`, etc labels, but `NamedKey` in the linked w3c spec doesn't. So the suggested name is less arbitrary than `Action`. The main issue I have with action that it implies that I need to perform it, but `NamedKey` just tells that these are a bunch of keysyms with pre-defined labels, there could be more of them than regular KeyCode. KeyCodes are all about physical keys, not about named keysyms, which is what we try to desribe.

Reply:
Renamed to `NamedKey`.